llvm/lib/LTO/ThinLTOCodeGenerator.cpp

```diff
@@ -971,9 +1302,24 @@ void ThinLTOCodeGenerator::run() {
   });
   // Prepare the resulting object vector
   assert(ProducedBinaries.empty() && "The generator should not be reused");
-  if (SavedObjectsDirectoryPath.empty())
+
+  // Need SavedObjectsDirectory for remote cache.
+  bool UseBufferAPI = SavedObjectsDirectoryPath.empty();
```

Comment:
Shouldn't `bool UseBufferAPI = SavedObjectsDirectoryPath.empty();` move after the following code that initializes `SavedObjectsDirectoryPath`?

Reply:
This is intended behavior. We need to force a `SavedObjectDirectory` when we use remote CAS as the place where the service pass file back, but we need to respect the original setting for whether pass the file or buffer back to the caller (indicated by whether there is an original setting on the object directory). I will add some comments to the code.

Reply:
To make sure I understand, the linker uses `SavedObjectsDirectoryPath` to indicate whether it wants to receive objects as files or as buffers?

Reply:
correct.